src/Components/TodoContainer.js

```diff
@@ -1,74 +1,89 @@
+    const [isAscending, setIsAscending] = useState(true); // Track sorting order
 
-  // New async function to fetch data from the API
-  async function fetchData() {
-    // Define options for the API request
-    const options = {
-      method: "GET",
-      headers: {
-        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
-      },
-    };
+    // Memoize the fetchData function using useCallback to ensure a stable reference between renders as long as its dependencies (in this case, isAscending) don't change.
 
-    // Construct the URL for the API
-    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;
+    const fetchData = useCallback(async () => {
+        // Define options for the API request
+        const options = {
+            method: "GET",
+            headers: {
+                Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+            },
+        };
 
-    try {
-      const response = await fetch(url, options);
+        // Construct the URL for the API with ascending or descending order
+        // Update the URL with sort parameters (field and direction)
+        const order = isAscending ? "asc" : "desc";
+        const sortField = "created"; // Change this to the desired sorting field
+        const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}?sort[0][field]=${sortField}&sort[0][direction]=${order}`;
 
-      if (!response.ok) {
-        console.log("Airtable error response:");
-        console.log(response);
-        throw new Error(`Error: ${response.status}`);
-      }
+        try {
+            const response = await fetch(url, options);
 
-      // Parse the response JSON
-      const data = await response.json();
+            if (!response.ok) {
+                console.log("Airtable error response:");
+                console.log(response);
+                throw new Error(`Error: ${response.status}`);
+            }
 
-      // Transform API data.records into todo objects
-      const todos = data.records.map((record) => ({
-        id: record.id,
-        title: record.fields.title,
-      }));
+            // Parse the response JSON
+            const data = await response.json();
 
-      // Update todoList and isLoading states
-      setTodoList(todos);
-      setIsLoading(false);
-    } catch (error) {
-      console.log(`Fetch error: ${error.message}`);
-    }
-  }
+            // Transform API data.records into todo objects
+            const todos = data.records.map((record) => ({
+                id: record.id,
+                title: record.fields.title,
+            }));
+
+            // Update todoList and isLoading states
+            setTodoList(todos);
+            setIsLoading(false);
+        } catch (error) {
+            console.log(`Fetch error: ${error.message}`);
+        }
+    }, [isAscending]);
```

Comment:
Nice job getting this backend sorting working! 

In order to see both backend sorting and frontend sorting in this assignment, can you please make it work like this?

- On initial component load, the `App`'s `fetchData` function always fetches the data sorted by `created`  ascending.
- Then every time the re-sort button is clicked, use that changed `isAscending` state to `Array.sort()` the `todoList` state before passing it to the `TodoList`

Reply:
doing this resorting on the frontend will also mean that button will re-sort instantly, rather than having to wait a moment for the backend to provide the data